schematic/store/synapse.py

```diff
@@ -395,85 +491,44 @@ def getDatasetManifest(
-            if any(censored):
-                # Try to use uncensored manifest first
-                not_censored=~censored
-                if any(not_censored):
-                    manifest_syn_id=manifest[not_censored]["id"][0]
-
-            #otherwise, use the first (implied only) version that exists
-            else:
-                manifest_syn_id = manifest["id"][0]
-
-
-            # if the downloadFile option is set to True
-            if downloadFile:
-                # enables retrying if user does not have access to uncensored manifest
-                while True:
-                    # pass synID to synapseclient.Synapse.get() method to download (and overwrite) file to a location
-                    try:
-                        if 'manifest_folder' in CONFIG['synapse'].keys():
-                            manifest_data = self.syn.get(
-                                manifest_syn_id,
-                                downloadLocation=CONFIG["synapse"]["manifest_folder"],
-                                ifcollision="overwrite.local",
-                            )
-                            break
-                        # if no manifest folder is set, download to cache
-                        else:
-                            manifest_data = self.syn.get(
-                                manifest_syn_id,
-                            )
-                            break
-                    # If user does not have access to uncensored manifest, use censored instead
-                    except(SynapseUnmetAccessRestrictions):
-                            manifest_syn_id=manifest[censored]["id"][0]
-                    
-                # Rename manifest file if indicated by user.
-                if newManifestName:
-                    if os.path.exists(manifest_data['path']):
-                        # Rename the file we just made to the new name
-                        new_manifest_filename = newManifestName + '.csv'
-                        new_manifest_path_name = manifest_data['path'].replace(manifest['name'][0], new_manifest_filename)
-                        os.rename(manifest_data['path'], new_manifest_path_name)
-
-                        # Update file names/paths in manifest_data
-                        manifest_data['name'] = new_manifest_filename
-                        manifest_data['filename'] = new_manifest_filename
-                        manifest_data['path'] = new_manifest_path_name
-
+            manifest_syn_id = self._get_manifest_id(manifest)
+            if downloadFile: 
+                md = ManifestDownload()
+                manifest_data = ManifestDownload.download_manifest(md, self.syn, manifest_id=manifest_syn_id, newManifestName=newManifestName, manifest_df=manifest)
+                if manifest_data == "":
+                    logger.debug(f"No manifest data returned. Please check if you have successfully downloaded manifest: {manifest_syn_id}")
```

Comment:
should this be "logger.debug" or should this be a warning? I think most of the time when "manifest_data" returns an empty string, it usually indicates there's something wrong with the download_manifest function. Downstream function usually tries to extract information from manifest_data, and if it returns an empty string, usually the downstream function will break.

Reply:
If this breaks the downstream code, maybe it should be an error.  What is your rationale about not raising the error here?

Reply:
hmm Thanks Tom. I think about this issue a bit more, and I think it would make more sense to turn it to logger.error...Previously, I was thinking that since we initialized `manifest_data` as an empty string in the beginning (see [here](https://github.com/Sage-Bionetworks/schematic/blob/05d2dede44e31121067cecdc1b2a9313860de7e1/schematic/store/synapse.py#:~:text=%22%22%22-,manifest_data%20%3D%20%22%22,-%23%20get%20a%20list)), then it might make sense to "allow" returning an empty string back? And to be honest, i am not sure if we should always: 
* raise an error when the request failed
* Or return an empty value so that the request would go through without raising an error. 
I see a mix in our current code base. But I think if the downstream code does not accept an empty value, maybe we could just catch the error here and make it simpler.

Reply:
I think that depends on how the downstream code uses it.  If you don't want to break your downstream code, then it makes sense to catch the error here.  

You can also add a `# TODO` to potentially revisit.

Reply:
Thanks Tom. This is a good advice. I will make a comment and remind myself to revisit.